open_rarity/rarity_ranker.py

```diff
@@ -1,14 +1,22 @@
 import math
+from open_rarity.models.collection import Collection
+from open_rarity.models.token import Token
+from open_rarity.models.token_rarity import TokenRarity
+
+from open_rarity.scoring.scorer import Scorer
 
 
 class RarityRanker:
     """This class is used to rank a set of tokens given their rarity scores."""
 
+    default_scorer = Scorer()
+
     @staticmethod
-    def rank_tokens(*, token_id_to_scores: dict[str, float]) -> dict[str, int]:
-        """Ranks a set of tokens, identified uniquely by a string, based on
-        the scores provided. The ranking is done assuming @token_id_to_scores
-        contains all tokens for a given cohort/collection to rank tokens for.
+    def rank_collection(
+        collection: Collection, scorer: Scorer = default_scorer
+    ) -> Collection:
```

Comment:
Commentary on the interface:
- I like that we take in scorer so that ranking + scoring can happen in a single request by user (vs. two separate requests)
- I don't think we should modify the inputs and return a modified Collection. Perhaps we can instead return a new object called `RankedCollection`
- Might be nice for user to specify the order of returned tokens (if we choose to return it in array format) 
e.g. rank_collection(collection, sort: [:rank | :identifier ], order: :asc)

Reply:
One example/option:
```
def RankedCollection:
   # If we don't provide a sort/order option, this can just match same order of collection.tokens
   token_ranks: list[TokenRarity]
```

Reply:
cc @theelderbeever for thoughts

Reply:
Maybe for simplicity we don't support ordering/sorting and just do something similiar to proposed interfae where we just match same order

Reply:
We could skip inputs part and augment tokens only with rarity information. sorting can happen outside of the request.

Reply:
The user can always set a specific ordering of `collection.tokens` themselves and we just follow it.

Reply:
For context, taking the following use case a user trying to integrate with library and calculating ranking + scoring to integrate into their own system

```
        # return a dictionary of token ID to the corresponding rank/rarity
        token_rarities = OpenRarityRanker.rank_collection(collection=or_collection)
        return {
            token.token_identifier: RankScore(token_rarities[i]) for idx, token in enumerate(or_collection.tokens)
        }
```

Reply:
cool  , curios what others think on the lib interface - cc @theelderbeever @dadashi

Reply:
I've updated the class and came up with mix of both worlds take a look at new rev. The idea is to add token link to the TokenRarity class without replicating collection container, as caller has it on their side anyways.

Reply:
fixed